test/integration/testStreamRows.js

```diff
@@ -372,4 +373,107 @@ describe('Test Stream Rows API', function ()
       }
     });
   });*/
+
+});
+
+describe('testHighWaterMark', function ()
+{
+  this.timeout(300000);
+
+  before(function (done)
+  {
+    connection = testUtil.createConnection();
+    testUtil.connect(connection, done);
+  });
+
+  after(function (done)
+  {
+    testUtil.destroyConnection(connection, done);
+  });
+
+  var testingFunc = function (highWaterMark, expectedRowCount, callback)
+  {
+    async.series(
+      [
+        function (callback)
+        {
+          // select table with row count equal to expectedRowCount
+          var statement = connection.execute({
+            sqlText: `SELECT seq8() FROM table(generator(rowCount => ${expectedRowCount}));`,
+            streamResult: true,
+            complete: function ()
+            {
+              var actualRowCount = 0;
+              var rowIndex;
+
+              var stream = statement.streamRows();
+              stream.on('error', function (err)
+              {
+                callback(err);
+              });
+              stream.on('readable', function ()
+              {
+                rowIndex = 0;
+
+                while (this.read() !== null)
+                {
+                  actualRowCount++;
+                  rowIndex++;
+                }
+
+                // assert the amount of rows read per loop never exceeds the highWaterMark threshold
+                assert.ok(rowIndex <= highWaterMark);
+              });
+              stream.on('end', function ()
+              {
+                // assert the total number of rows is equal to the specified row count
+                assert.strictEqual(actualRowCount, expectedRowCount);
+
+                callback();
+              });
+            }
+          });
+        }
+      ],
+      callback
+    );
+  };
+
+  const highWaterMarkValue = 10; // default parameter value is 10 (based on PARAM_ROW_STREAM_HIGH_WATER_MARK)
```

Comment:
Currently the [highWaterMark parameter](https://github.com/snowflakedb/snowflake-connector-nodejs/blob/master/lib/connection/connection_config.js#L846) is not an external parameter so only the default value is tested

Only parameters that are external can be modified through the connection options: https://github.com/snowflakedb/snowflake-connector-nodejs/blob/master/lib/connection/connection_config.js#L743

Here's the [timeout parameter](https://github.com/snowflakedb/snowflake-connector-nodejs/blob/master/lib/connection/connection_config.js#L813) which is an external parameter as a reference